### ISSUE TITLE
flutter doctor --android-licenses fails with recent Android SDK tools

### ISSUE THREAD

**Issue by hapasa** [flutter/flutter#137022]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

```console
 flutter doctor --android-licenses
Error: LinkageError occurred while loading main class com.android.sdklib.tool.sdkmanager.SdkManagerCli
        java.lang.UnsupportedClassVersionError: com/android/sdklib/tool/sdkmanager/SdkManagerCli has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
Android sdkmanager tool was found, but failed to run
(/home/harri/Android/Sdk/cmdline-tools/latest/bin/sdkmanager): "exited code 1".
Try re-installing or updating your Android SDK,
visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed
instructions.
```

### Expected results

Licenses should be displayed for acceptance.

Workaround is to downgrade command line tools in Android SDK Manager to 10.0 and then run flutter doctor --android-licenses again.

### Actual results

```console
Error: LinkageError occurred while loading main class com.android.sdklib.tool.sdkmanager.SdkManagerCli
        java.lang.UnsupportedClassVersionError: com/android/sdklib/tool/sdkmanager/SdkManagerCli has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
```
There is a workaround

### Code sample

flutter doctor --android-licenses

### Screenshots or Video

<details>
<summary>Android Studio SDK Manager setting</summary>

![sdk_manager](https://github.com/flutter/flutter/assets/8208561/d885d0a1-2690-467c-9391-e6d40a79ca03)

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.8, on Ubuntu 22.04.3 LTS 6.2.0-34-generic, locale
    en_US.UTF-8)
[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more
      details.
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2022.1)
[✓] Proxy Configuration
[✓] Connected device (2 available)
[✓] Network resources

! Doctor found issues in 1 category.

```

</details>

**Comment by huycozy** [comment]
Hi @hapasa 
Please check [this](https://github.com/flutter/flutter/issues/132142#issuecomment-1670072536) and the following comment to see if it helps.

**Comment by sittingbool** [comment]
Same here, reinstalling the SDK or reinstalling it as Version 11 and renaming it to latest did not work as well

**Comment by huycozy** [comment]
@sittingbool Could you share the output of `flutter doctor -v` and `java --version`? (cc @hapasa for sharing this info as well)

**Comment by sittingbool** [comment]
flutter doctor -v
[✓] Flutter (Channel stable, 3.13.8, on macOS 13.5 22G74 darwin-x64, locale de-DE)
    • Flutter version 3.13.8 on channel stable at /Users/richardhabermann/Developer/TOOLS/FlutterSDK/flutter_macos_3.13.8-stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6c4930c4ac (8 days ago), 2023-10-18 10:57:55 -0500
    • Engine revision 767d8c75e8
    • Dart version 3.1.4
    • DevTools version 2.25.0

[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/richardhabermann/Library/Android/sdk
    • Platform android-34-ext8, build-tools 34.0.0
    • ANDROID_HOME = /Users/richardhabermann/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/richardhabermann/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 76.1.4
    • Dart plugin version 232.10072.19

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.5 22G74 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 118.0.5993.88

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

**Comment by sittingbool** [comment]
java --version
openjdk 21.0.1 2023-10-17
OpenJDK Runtime Environment (build 21.0.1+12-29)
OpenJDK 64-Bit Server VM (build 21.0.1+12-29, mixed mode, sharing)

**Comment by huycozy** [comment]
@sittingbool As I can see from your flutter doctor output, Flutter is using Java on `/Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java` instead of on the system, it's using Java 11. 

Could you share the output of this (with verbose)?
```
flutter doctor --android-licenses -vv
```

**Comment by SageMik** [comment]
I also encountered the same issue in Windows, which was resolved by **uninstalling and reinstalling Android Studio, upgrading from version 2022.2 to the latest version 2022.3**. The version of Java that comes with Android Studio has also been upgraded to _Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)_. So I can successfully run `flutter doctor --android-licenses` now.
It seems that Flutter uses the JDK provided by Android Studio by default. Even if JAVA_HOME is set, it will not select another version.
